src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java

```diff
@@ -208,6 +208,29 @@ public static int lengthExpandedTabs(String inputString,
     }
 
     /**
+     * Returns the length of a String prefix with tabs compressed.
+     *
+     * @param inputString
+     *            the input String
+     * @param toIdx
+     *            index in string (exclusive) where the calculation stops
+     * @param tabWidth
+     *            the distance between tab stop position.
+     * @return the length of string.substring(0, toIdx) with tabs compressed.
+     */
+    public static int lengthCompressedTabs(String inputString,
```

Comment:
Can we use lengthExpandedTabs to have the logic which is similar to AbstractCheck#log?

FYI:

```java
len = (len / tabWidth + 1) * tabWidth;
```

is equal to

```java
len = len + tabWith; 
```

which is equal to
```java
len += tabWith
```

Reply:
> len = (len / tabWidth + 1) * tabWidth;

looks like `len = (len / tabWidth + 1) * tabWidth` is not same as `len += tabWith`
you can see `CommonUtilsTest.testLengthExpandedTabs` is failing

example: len = 2 and tabWidth = 8
len = (len / tabWidth + 1) * tabWidth  ==  8
len += tabWith  == 10

Reply:
> `len = (len / tabWidth + 1) * tabWidth` is not same as `len += tabWith`

This because simplification expects real numbers with decimal.
Code is secretly written/compiled like `len = int(len / tabWidth + 1) * tabWidth` which drops the decimal.

Reply:
> This is because len and tabWidth are integers.

It does not mean that we cannot apply maths rules to simplify fractions. The problem is that the expression in incorrect. It should be
```
len = Math.floor(len / tabWidth + 1) * tabWidth
```

as you said.

Reply:
@MEZk As I also stated, the code written is already like that as integers don't have decimals. Since `len` and `tabWidth` are integers, there will never be a decimal from division so `Math.floor` is completely unnecessary.
Even implementation of method won't accept integers: https://docs.oracle.com/javase/7/docs/api/java/lang/Math.html#floor(double)

Reply:
```len = (len / tabWidth + 1) * tabWidth```

This expression still looks confusing.

Reply:
@timurt 
Can you prove that we cannot use the existing method with this logic
```
len = (len / tabWidth + 1) * tabWidth
```

Reply:
@MEZk 
You are right, I was trying to reimplement the same logic - it is not correct I guess

@MEZk @rnveach 
But I do not understand why we have `len = Math.floor(len / tabWidth + 1) * tabWidth` instead of `len += tabWith`
Can you explain?